sycl/doc/UsersManual.md

```diff
@@ -1,63 +1,120 @@
+Options marked as [EXPERIMENTAL] are expected to be used only in limitted cases
+and not recommended to use in production environment.
+
+## Generic options
 
 **`-fsycl`**
 
-    General enabling option for SYCL compilation mode.  This option enables
-    offload compilation for a given target using the `-fsycl-targets` option.
-    When the `-fsycl-targets` option is not provided, the default triple is
-    `spir64-unknown-unknown-sycldevice`.
+    General enabling option for SYCL compilation and linking mode. List of
+    targets can be specified with `-fsycl-targets`. This is fundamental option
+    for any SYCL compilation. All other SYCL specific options require it.
+
+**`-fsycl-targets=<T1>[,...,<Tn>]`**
+
+    Tells the compiler to generate code for specified devices. T is a target
+    triple device name. You can specify more than one target, comma separated.
+    The following triples are supported by default:
+    * spir64-unknown-unknown-sycldevice - this is the default generic SPIR-V
+      target;
+    * spir64_x86_64-unknown-unknown-sycldevice - generate code ahead of time
+      for x86_64 CPUs;
+    * spir64_fpga-unknown-unknown-sycldevice - generate code ahead of time for
+      Intel FPGA;
+    * spir64_gen-unknown-unknown-sycldevice - generate code ahead of time for
+      Intel Processor Graphics;
+    Available in special build configuration:
+    * nvptx64-nvidia-cuda-sycldevice - generate code ahead of time for CUDA
+      target;
+
+## Language options
+
+**`-sycl-std=<value>`** [EXPERIMENTAL]
 
-**`-fsycl-targets=<value>`**
+    SYCL language standard to compile for. Possible values:
+    * 121 - SYCL 1.2.1
+    * 2020 - SYCL 2020
+    It doesn't guarantee specific standard compliance, but some selected
+    compiler features change behavior.
+    It is under development and not recommended to use in production
+    environment.
+    Default value is 2020.
 
-    A comma separated list of triples to specify the device target(s) to
-    generate code for.  This option is only valid when used with `-fsycl`.
+**`-f[no-]sycl-unnamed-lambda`**
 
-### Target toolchain options.
+    Enables/Disables unnamed SYCL lambda kernels support.
+    Disabled by default.
 
-**`-Xsycl-target-backend=<triple> <arg>`**
+## Optimization options
 
-    Pass <arg> to the SYCL based backend identified by <triple>.
+**`-f[no-]sycl-early-optimizations`**
 
-**`-Xsycl-target-backend <arg>`**
+    Enables (or disables) intermediate representation optimization pipeline
+    before translation to SPIR-V. Have effect only if optimizations are turned
+    on by standard compiler options (-O1 or higher).
+    Enabled by default.
 
-    Pass <arg> to the SYCL based target backend.
+**`-f[no-]sycl-dead-args-optimization`**
 
-**`-Xsycl-target-frontend=<triple> <arg>`**
+    Enables (or disables) LLVM IR dead argument elimination pass to remove
+    unused arguments for the kernel functions before translation to SPIR-V.
+    Disabled by default.
```

Comment:
some targets (PTX) don't use SPIRV, need to specify option's effect for them as well. @bader, do you know if this has effect for PTX?

Reply:
IIRC, this optimization impacts only SPIR-targeted code. Relevant issue - https://github.com/intel/llvm/issues/2359.

Reply:
Thanks. Then this could be rephrased as "... remove unused arguments for the kernel functions before translation to SPIR-V.
Has no effect on non-SPIRV targets. Disabled by default."

Reply:
Added.